src/Graphing.hs

```diff
@@ -279,19 +294,7 @@ deeps = Graphing . AM.vertices . map Node
 --
 -- __Unfold does not work for recursive inputs__
 unfold :: forall dep res. Ord res => [dep] -> (dep -> [dep]) -> (dep -> res) -> Graphing res
-unfold seed getDeps toDependency = directs directNodes <> edges builtEdges
-  where
-    directNodes :: [res]
-    directNodes = map toDependency seed
-
-    builtEdges :: [(res, res)]
-    builtEdges = map (bimap toDependency toDependency) (edgesFrom seed)
-
-    edgesFrom :: [dep] -> [(dep, dep)]
-    edgesFrom nodes = do
-      node <- nodes
-      let children = getDeps node
-      map (node,) children ++ edgesFrom children
+unfold = unfold' directs
```

Comment:
I'm actually not sure I'm seeing what is so bad here. If someone passes in an infinitely large seed list then it would never terminate. Likewise if you do an unfold with a function that has a code-path which might never produce a stopping condition. The main difference is that a graph cannot be generated lazily like a list can. I guess for me these scenarios feel expected rather than being a surprising foot-gun.

One thing that might help is to specify that the stopping condition for the unfold is producing an empty list, and that every output of it should get closer to producing `[]` when it is called again with that output as an input.